src/main/scala/tilelink/Parameters.scala

```diff
@@ -318,6 +357,44 @@ case class TLClientPortParameters(
     }
   }
 
+  private def emitHelper(
```

Comment:
it is very similar, but it instead looks at visibility instead of address

Reply:
That could be abstracted into a function argument. I'm concerned about copying code that looks very similar because, as an example, the existence of `groupByIntoSeq` in this method. I added that to `TLManagerPortParameters.supportHelper` because it was generating non-deterministic code. If we had had 2 functions, then I likely would've missed one. 

I'm not certain that it *should* be abstracted or what the new utility function would be called, thoughts @terpstra?